packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,356 @@
+
+    // beneficiary of the amount
+    address public mBeneficiary;
+
+    // durations in secs. of one period
+    uint256 public mVestingPeriodSec;
+
+    // timestamps for start and cliff starting points. Timestamps are expressed in UNIX time, the same units as block.timestamp.
+    uint256 public mCliffStartTime;
+    uint256 public mVestingStartTime;
+
+    // indicates if the contract is revokable
+    bool public mRevocable;
+
+    // revoking address and refund destination
+    address public mRefundDestination;
+    address public mRevoker;
+
+    // indicates how much of the vested amount has been released for withdrawal (i.e. withdrawn)
+    uint256 public mCurrentlyReleased;
+
+    // indicates if the vesting has been revoked. false by default
+    bool public mRevoked;
+
+    // the time at which the revocation has taken place
+    uint256 public mRevokeTime;
+
+    /**
+     * @notice A constructor for initialising a new instance of a Vesting Schedule contract
+     * @param beneficiary address of the beneficiary to whom vested tokens are transferred
+     * @param vestingAmount the amount that is to be vested by the contract
+     * @param vestingCliff duration in seconds of the cliff in which tokens will begin to vest
+     * @param vestingStartTime the time (as Unix time) at which point vesting starts
+     * @param vestingPeriodSec duration in seconds of the period in which the tokens will vest
+     * @param vestAmountPerPeriod the vesting amound per period where period is the vestingAmount distributed over the vestingPeriodSec
+     * @param revocable whether the vesting is revocable or not
+     * @param revoker address of the person revoking the vesting
+     * @param refundDestination address of the refund receiver after the vesting is deemed revoked
+     */
+    constructor (address beneficiary,
+                uint256 vestingAmount,
+                uint256 vestingCliff,
+                uint256 vestingStartTime,
+                uint256 vestingPeriodSec,
+                uint256 vestAmountPerPeriod,
+                bool    revocable,
+                address revoker,
+                address refundDestination) public {
+
+        // do some basic checks
+        require(vestingAmount > 0, "Amount must be positive");
+        require(beneficiary != address(0), "Beneficiary is the zero address");
+        require(refundDestination != address(0), "Refund destination is the zero address");
+        require(vestingCliff <= vestingPeriodSec, "Vesting cliff is longer than duration");
+        require(vestingPeriodSec > 0, "Vesting period is 0 s.");
+        require(vestAmountPerPeriod <= vestingAmount, "Vesting amount per period is greater than the total vesting amount");
+        require(vestingStartTime.add(vestingCliff) > block.timestamp, "Final time is before current time");
+
+        //make the vesting instance an account
+        bool accountCreated = getAccounts().createAccount();
```

Comment:
We don't necessarily need to do this, we can provide a wrapper and let the beneficiary choose to do this

Reply:
done